### PR TITLE
Blob block error handling

### PR DESCRIPTION
**Pull request by pomeroy3**
Add error log statements intended to clarify exceptions due to incorrect URLs, bad authentication credentials, bad permissions, and missing containers and blobs.

Also adds a retry_total parameter that controls how many times blob API calls will be retried when they fail.

### CODE REVIEW

File: gr-azure-software-radio/README.md [hunk 1]
```diff
@@ -55,27 +55,31 @@ Run the QA tests with any of the following methods:
...
     ```
     python -m pytest qa_*
-    ```    
+    ```
 
 ### Running the Integration Tests
-To run the integration tests for the blob blocks, you must first create a storage account on Azure 
-and obtain the connection string. Next, export this connection string into an environment variable 
-named "AZURE_STORAGE_CONNECTION_STRING". 
+To run the integration tests for the blob blocks, you must first create a storage account on Azure
```
Comment (jo-rivera) [inline gr-azure-software-radio/README.md:61]: we need to update the Table of Contents "running the tests" to link to the unit test and integration test sections

File: gr-azure-software-radio/python/blob_sink.py [hunk 2]
```diff
@@ -38,13 +41,15 @@ class BlobSink(gr.sync_block):
...
+    Retry Total: Total number of Azure API retries to allow before throwing an exception. Higher
+        numbers make the block more resilient to intermittent failures, lower numbers help to
+        more quickly iteratively debug authentication and permissions issues.
     """
     # pylint: disable=too-many-arguments, too-many-instance-attributes, arguments-differ, abstract-method
 
     def __init__(self, authentication_method: str = "default", connection_str: str = None,
                  url: str = None, container_name: str = None, blob_name: str = None,
-                 block_len: int = 500000, queue_size: int = 4):
-        # pylint: disable=no-member
```
Comment (jo-rivera) [inline gr-azure-software-radio/python/blob_sink.py:47]: do we need the pylint: disable=no-member?

File: gr-azure-software-radio/python/blob_sink.py [hunk 3]
```diff
@@ -80,6 +88,9 @@ def __init__(self, authentication_method: str = "default", connection_str: str =
 
         self.blocks_per_commit = 2
```
Comment (jo-rivera) [inline gr-azure-software-radio/python/blob_sink.py:89]: is this getting used? is it always 2?

File: gr-azure-software-radio/python/blob_source.py [hunk 4]
```diff
@@ -34,12 +38,14 @@ class BlobSource(gr.sync_block):
...
     Blob Name: The name of the block blob to read from.
+    Retry Total: Total number of Azure API retries to allow before throwing an exception. Higher
+        numbers make the block more resilient to intermittent failures, lower numbers help to
+        more quickly iteratively debug authentication and permissions issues.
     """
     # pylint: disable=too-many-arguments, too-many-instance-attributes, arguments-differ, abstract-method
     def __init__(self, authentication_method: str = "default", connection_str: str = None,
                  url: str = None, container_name: str = None, blob_name: str = None,
-                 queue_size: int = 4):
-        # pylint: disable=no-member
```
Comment (jo-rivera) [inline gr-azure-software-radio/python/blob_source.py:42]: same as above with pylint: disable=no-member

File: gr-azure-software-radio/python/blob_sink.py (hunk 3 above)
Comment (pomeroy3) [reply gr-azure-software-radio/python/blob_sink.py:89]: nope, I can pull it

File: gr-azure-software-radio/python/blob_sink.py (hunk 2 above)
Comment (pomeroy3) [reply gr-azure-software-radio/python/blob_sink.py:47]: I didn't see any pylint errors in the latest build with it gone. Think we still need it?

**Review by markosly — APPROVED** [review]
Except for Johanna's questions this looks good to me.

File: gr-azure-software-radio/python/blob_sink.py (hunk 2 above)
Comment (jo-rivera) [reply gr-azure-software-radio/python/blob_sink.py:47]: we don't need it then :)

File: gr-azure-software-radio/README.md (hunk 1 above)
Comment (pomeroy3) [reply gr-azure-software-radio/README.md:62]: This was a great catch btw

**Pull request merged**